prometheus/go_collector_go117.go

```diff
@@ -303,37 +317,38 @@ type batchHistogram struct {
 	sum     float64 // Used if hasSum is true.
 }
 
-func newBatchHistogram(desc *Desc, hasSum bool) *batchHistogram {
-	h := &batchHistogram{desc: desc, hasSum: hasSum}
+func newBatchHistogram(desc *Desc, buckets []float64, hasSum bool) *batchHistogram {
+	h := &batchHistogram{
+		desc:    desc,
+		buckets: buckets,
+		counts:  make([]uint64, len(buckets)-1),
```

Comment:
Maybe I missed something, but why -1?

Reply:
Ah, ok found it, it similar to this in metrics:

> 	// Given N buckets, Count[n] is the weight of the range
>	// [bucket[n], bucket[n+1]), for 0 <= n < N.

Worth adding similar comment?

Reply:
done